### PR TITLE
Created Connection Page

### PR DESCRIPTION
**Pull request by Shea-Scarlet**
Added temporary api.ts for testing

### CODE REVIEW

**Comment by Shea-Scarlet** [comment]
Websockets are online!

File: backend/.babelrc
Comment (190n) [inline backend/.babelrc:1]: Why's this here? I don't think we use Babel on the backend.

File: frontend/package.json [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
+
+  "name": "kakaw",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
+    "test": "jest --coverage"
```
Comment (190n) [inline frontend/package.json]:
```suggestion
    "test": "jest --coverage",
```

npm doesn't parse this without the comma

File: frontend/package.json [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
+  "devDependencies": {
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^14.0.0",
+    "@testing-library/user-event": "^14.4.3",
+    "@types/jest": "^29.5.0",
+    "@types/testing-library__jest-dom": "^5.14.5",
+    "eslint-config-prettier": "^8.8.0",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^29.5.0",
+    "jest-environment-jsdom": "^29.5.0"
```
Comment (190n) [inline frontend/package.json]:
```suggestion
    "jest-environment-jsdom": "^29.5.0",
```

File: frontend/pages/[gameId]/[playerId].tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { WEBSOCKET_BASE_URL } from '/Users/gealoro/kakaw/frontend/lib/api';
```
Comment (190n) [inline frontend/pages/[gameId]/[playerId].tsx]:
```suggestion
import { WEBSOCKET_BASE_URL } from '@/lib/api';
```

The current import only works for you. `@` refers to the root of the package (in this case `frontend`).

File: frontend/pages/[gameId]/[playerId].tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    //temp values
+    const gameId = '123';
+    const playerId = '456';
+  
+    return (
+      <Link href={`/connect?gameId=${gameId}&playerId=${playerId}`}>
+        <a>Connect</a>
+      </Link>
+    );
+  }
```
Comment (190n) [inline frontend/pages/[gameId]/[playerId].tsx]:
```suggestion
```

Dead code

File: frontend/pages/[gameId]/[playerId].tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      console.log('WebSocket connection established.');
+    }
+  });
+
+  return (
+    <div>Hello WebSockets!</div>
+  );
+}
+/*/
+
```
Comment (190n) [inline frontend/pages/[gameId]/[playerId].tsx]:
```suggestion
```

File: frontend/pages/[gameId]/[playerId].tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+function ConnectPage() {
+
+  useWebSocket(WS_URL, {
```
Comment (190n) [inline frontend/pages/[gameId]/[playerId].tsx]:
```suggestion
	const router = useRouter();
	const { gameId, playerId } = router.query as {
		gameId: string;
		playerId: string;
	};

	const url = new URL('/connect', WEBSOCKET_BASE_URL);
	url.searchParams.set('gameId', gameId);
	url.searchParams.set('playerId', playerId);

	useWebSocket(url.href, {
```
`WEBSOCKET_BASE_URL` is just the hostname/port we should connect to. So right now it's `ws://localhost:8080` but maybe later we would change it to `wss://kakaw.com` or wherever we might end up hosting this app. You still need to specify `/connect` as well as the query parameters for the game and player IDs.

File: frontend/pages/[gameId]/[playerId].tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    onOpen: () => {
+      console.log('WebSocket connection established.');
+    }
+  });
+
+    const router = useRouter();
+    const { gameId, playerId } = router.query;
+
```
Comment (190n) [inline frontend/pages/[gameId]/[playerId].tsx]:
```suggestion
```

I moved these up above the `useWebSocket` call.

**Review by 190n — CHANGES_REQUESTED** [review]
Good work but a couple things need fixing. Read over my changes and then also please format all the code before merging.

**Comment by 190n** [comment]
Also please give your branches descriptive names.

**Review by 190n — APPROVED** [review]

**Pull request merged**